src/lair/toplevel.rs

```diff
@@ -879,3 +932,118 @@ impl<F: Field + Ord> OpE<F> {
         }
     }
 }
+
+impl<F: Field + Ord> Func<F> {
+    fn split(&self, groups: &FxIndexSet<ReturnGroup>) -> FxIndexMap<ReturnGroup, Func<F>> {
```

Comment:
`split` would really benefit from a docstring with a high level explanation of how "splitting" works